### ISSUE TITLE
Remove predicate functions - use `filter_deployments()`

### ISSUE THREAD

**Issue by damianooldoni** [inbo/camtraptor#316]
Once #315 is completed, we can replace the predicates with filter functions. Here below the most important steps:

- [x] Remove predicate related functions
- [x] Replace the use of predicate function with `filter_deployments()`
- [x] Update function documentation
- [ ] Update vignette #320 
- [x] Remove tests

**Comment by PietrH** [comment]
Functions should not do any filtering at all internally, any filtering should happen before and be piped in. 

So lines using predicates can just be removed.

**Comment by PietrH** [comment]
We should make a new vignette about the filtering functions : _how to read data_, _how to manipulate camtrapdp objects_

**Comment by PietrH** [comment]
There are still references to `mica` in my branch:

https://github.com/inbo/camtraptor/blob/a10adc1adc5ef8d2fc8b40b58894a8225f03de37/R/get_n_species.R#L14

Probably should remove these, I don't want to do this in the same branch

**Comment by PietrH** [comment]
I should also scan the package of the word `filter` in the documentation and examples

**Comment by damianooldoni** [comment]
Based on good practices, we must make these functions defunct, instead of just removing them.

**Comment by damianooldoni** [comment]
Fixed by PR #331. The task of updating the vignette has been already transferred to a separate issue.
